### PR TITLE
Retry decorator

### PR DESCRIPTION
**Pull request by SydneyDlh**
created a retry decorator that retries the function if it throws an error as requested in this [PR](https://github.com/app-sre/qontract-reconcile/pull/154#pullrequestreview-269575846)

### CODE REVIEW

File: utils/retry.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+        @wraps(f)
+        def f_retry(*args, **kwargs):
+            attempt = 0
+            attempts = 3
+            while True:
+                try:
+                    return f(*args, **kwargs)
+                except Exception as e:
+                    attempt += 1
```
Comment (jmelis) [inline utils/retry.py]: we can drop this line if we change the `while` above

File: utils/retry.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+                    if attempt == attempts:
```
Comment (jmelis) [inline utils/retry.py]: and this can we change to:

`if attempt > max_attempts - 1`

File: utils/retry.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+                        raise e
+                    else:
```
Comment (jmelis) [inline utils/retry.py]: no need for this `else`, just the `time.sleep`  after the `if` statement above

File: utils/retry.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+# source: https://www.calazan.com/retry-decorator-for-python-3/
+
+
+def retry(max_attempts=3):
+
+    def deco_retry(f):
+
+        @wraps(f)
+        def f_retry(*args, **kwargs):
+            attempt = 0
```
Comment (jmelis) [inline utils/retry.py]: ah, sorry, I missed this... this line is not needed anymore

**Comment by maorfr** [comment]
thanks for this as well @SydneyDlh!

**Pull request merged**